File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area = 0;
        int i = 0;
        int j = height.size() - 1;
        int res = 0;
        while (i < j) {
            int area = min(height[i], height[j]) * (j - i);
            res = max(res, area);
            if (height[i] <= height[j]) {
                i++;
            } else {
                j--;
            }
        }
        return res;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let maxArea = 0
    let area = 0
    let i = 0;
    let j = height.length - 1
    while (i < j) {
        area = Math.min(height[i], height[j]) * (j - i)
        maxArea = Math.max(area, maxArea)
        if (height[i] < height[j]) {
            i++

        } else {
            j--
        }
    }

    return maxArea

};